packages/ra-ui-materialui/src/input/DateInput.tsx

```diff
@@ -131,20 +133,15 @@ export const DateInput = ({
             return;
         }
 
+        const newValue = localInputRef.current.value;
         // To ensure users can clear the input, we check its value on blur
         // and submit it to react-hook-form
-        const newValue =
-            localInputRef.current.valueAsDate !== undefined &&
-            localInputRef.current.valueAsDate !== null &&
-            !isNaN(new Date(localInputRef.current.valueAsDate).getTime())
-                ? parse
-                    ? parse(localInputRef.current.valueAsDate)
-                    : getStringFromDate(localInputRef.current.valueAsDate)
-                : parse
-                  ? parse(localInputRef.current.value)
-                  : getStringFromDate(localInputRef.current.value);
-
-        if (newValue !== field.value) {
+        const isNewValueValid =
+            newValue === '' ||
+            (localInputRef.current.valueAsDate != null &&
+                !isNaN(new Date(localInputRef.current.valueAsDate).getTime()));
+
+        if (isNewValueValid) {
```

Comment:
Shouldn't you also keep the check `newValue !== field.value`?
This would avoid calling `field.onChange()` if the value has not changed, and the input was only burred.

Reply:
RHF handles this itself